### PR TITLE
eZ Platform v3 compatibility

### PR DESCRIPTION
**Pull request by kmadejski**
This PR brings compatibility for eZ Platform v3. Due to introduced changes it should be released under a different major tag as it's no longer compatible with eZ Platform v2.x and lower.

### CODE REVIEW

**Comment by kmadejski** [comment]
@adamwojs commits from https://github.com/ezsystems/ezplatform-automated-translation/pull/12/commits/239ba3d0ca42a5ac40f4540f2420dbabb160bfcb contain changes you have requested. I did it according to my best knowledge, but some things might still be missing though. It must be released as under a new major version as there are changes (added missing return types) on the interfaces. 

@lserwatka could we have a QA certification on this one, please?

**Comment by kmadejski** [comment]
@bdunogier / @damianz5 any feedback on this one, please?

File: lib/Translator.php [hunk 1]
```diff
@@ -1,76 +1,57 @@
...
 use eZ\Publish\API\Repository\Values\Content\Content;
 use eZ\Publish\API\Repository\Values\ContentType\FieldDefinition;
 use eZ\Publish\Core\MVC\Symfony\Locale\LocaleConverterInterface;
 
-/**
- * Class Translator.
- */
 class Translator
 {
     use RepositoryAware;
```
Comment (arfaram) [inline lib/Translator.php:19]: @kmadejski The trait class `RepositoryAware.php` is removed. The `use` will cause exception

File: lib/TranslatorGuard.php [hunk 2]
```diff
@@ -29,25 +27,26 @@ class TranslatorGuard
 {
     use RepositoryAware;
```
Comment (arfaram) [inline lib/TranslatorGuard.php:28]: @kmadejski The trait class `RepositoryAware.php` is removed. The `use` will cause an exception

File: lib/Translator.php (hunk 1 above)
Comment (kmadejski) [reply lib/Translator.php:19]: That's right, just got it from QA as well. I'll fix it soon :)

**Comment by micszo** [comment]
Hi @kmadejski ! Thanks for fixing the RepositoryAware issue. 😎 

It's now possible to use this new feature on 3.0, below please find the non-blocking things I noticed.

On the UI the checkbox seems to need some more love, unless I'm missing some assets but I tried reloading translations, yarn etc.:
Screenshot 2020-06-12 at 09 00 07

In the browser console an error occurs when creating a new translation, it looks a bit different on Chrome and Firefox.

FF:
Screenshot 2020-06-12 at 09 02 25
`TypeError: mapping is undefined`

Chrome:
Screenshot 2020-06-12 at 09 03 04
`Uncaught TypeError: Cannot read property 'google' of undefined`

I'll let you know if I find anything else.

**Comment by micszo** [comment]
What about support for Landing Pages? 🙂 e.g. some text in a Code block. The checkbox is visible when adding a translation from Page Builder level but the magic doesn't happen.

**Comment by kmadejski** [comment]
@micszo unfortunately, it does not support Landing Pages but it could be a nice improvement 🙂

**Comment by micszo** [comment]
Ok, similarly the option is present when adding translations in Form Builder. 🙂 But here it could be only applicable to placeholder values I suppose.

**Comment by kmadejski** [comment]
@micszo I have identifier the root source of the problem you spot during recent tests. There is a problem with template overriding. This bundle tries to override one of the templates, precisely `admin/content/modal/add_translation.html.twig` ( https://github.com/ezsystems/ezplatform-automated-translation/blob/v3/bundle/Resources/views/themes/admin/content/modal/add_translation.html.twig) which is necessary to display a switcher (checkbox). 
Since this happens within the same theme, `admin`, the template precedence order is taken from bundles list order defined in `bundles.php`. I'm unsure if this behavior is completely correct, but it is what it is. To make it work, you need to change bundles order in `bundles.php` and move `EzSystems\EzPlatformAutomatedTranslationBundle\EzPlatformAutomatedTranslationBundle::class => ['all' => true],` before `EzSystems\EzPlatformAdminUiBundle\EzPlatformAdminUiBundle::class => ['all' => true],`. 

I have updated bundle's documentation to cover this trick in: https://github.com/ezsystems/ezplatform-automated-translation/pull/12/commits/1b7e543224b4b0d5fee006432ba9e4ff4d29be5f

**Comment by bdunogier** [comment]
I'm not surprised by the behaviour you describe, @kmadejski. I'm afraid it is "correct", as in "not a side effect", but it doesn't feel right from a user perspective. I don't know what we could do about it, but it would be great if we could find a better way than fiddling with bundles order.

**Comment by micszo** [comment]
@kmadejski changing bundles order resolves the problem on v3.0.6. 👍 
Should it also work for v3.1.0? 
With the same setup I can't get it to work on 3.1.

**Comment by kmadejski** [comment]
@micszo I was testing that on v3.1.0 and it seems to be alright. What is the issue you're facing?

**Comment by micszo** [comment]
Yesterday on v3.1.0 the order change had no effect for me but now on v3.1.1 it works fine. Not sure what was the issue.

**Comment by kmadejski** [comment]
@micszo Just guessing that it could be cache 😅  Thanks for testing!

**Pull request merged**